scripts/docker-helper/cht-docker-compose.sh

```diff
@@ -0,0 +1,442 @@
+#!/usr/bin/env bash
+
+# Helper script to get a docker-compose instance up and running.
+#
+# Uses docker-compose-developer.yml docker compose file to
+#
+#   * See if a .env file is available
+#   * Using info from .env, be aware of ports and storage names
+#   * make sure as much of the services in medic-os are up and running correctly
+#   * install valid certificate
+#   * show users status of instance
+#
+# See https://github.com/medic/cht-core/issues/7218 for more info
+
+# shellcheck disable=SC2046
+. $(dirname $0)/simple_curses.sh
+
+# todo maybe check to see if docker is running? avoid this error:
+# Error response from daemon: dial unix docker.raw.sock: connect: connection refused
+
+# todo MacOS doesn't print simple curses screens at  full width and is stuck at ~80 chars wide?
+
+get_lan_ip() {
+  ipInstalled=$(required_apps_installed "ip")
+  if [ -n "$ipInstalled" ]; then
+    lanAddress=127.0.0.1
+  else
+    # todo - some of these calls fail wien there's no network connectivity - output stuff it shouldn't:
+    #       Device "" does not exist.
+    routerIP=$(ip r | grep default | head -n1 | awk '{print $3}')
+    subnet=$(echo "$routerIP" | cut -d'.' -f1,2,3)
+    if [ -z $subnet ]; then
+      subnet=127.0.0
+    fi
+    lanInterface=$(ip r | grep $subnet | grep default | head -n1 | cut -d' ' -f 5)
+    lanAddress=$(ip a s "$lanInterface" | awk '/inet /{gsub(/\/.*/,"");print $2}' | head -n1)
+    if [ -z "$lanAddress" ]; then
+      lanAddress=127.0.0.1
+    fi
+  fi
+  echo "$lanAddress"
+}
+
+get_local_ip_url(){
+  lanIp=$1
+  cookedLanAddress=$(echo "$lanIp" | tr . -)
+  url="https://${cookedLanAddress}.my.local-ip.co:${CHT_HTTPS}"
+  echo "$url"
+}
+
+required_apps_installed(){
+  error=''
+  appString=$1
+  IFS=';' read -ra appsArray <<<"$appString"
+  for app in "${appsArray[@]}"; do
+    if ! command -v "$app" &>/dev/null; then
+      error="${app} ${error}"
+    fi
+  done
+  echo "${error}"
+}
+
+port_open(){
+  ip=$1
+  port=$2
+  # todo - macos prints this on screen. nothing should be shown (like on ubuntu)
+  # Connection to 127.0.0.1 port 8443 [tcp/pcsync-https] succeeded!
+  nc -z "$ip" "$port"
+  echo $?
+}
+
+has_self_signed_cert() {
+  # todo - when there's no connectivity, this fails w/ DNS
+  # curl: (6) Could not resolve host: 127-0-0-1.my.local-ip.co
+
+  # todo - macos returns this error some times
+  # curl: (35) LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to 127-0-0-1.my.local-ip.co:8443
+  url=$1
+  curl --insecure -vvI "$url" 2>&1 | grep -c "self signed certificate"
+}
+
+cht_healthy(){
```

Comment:
Docker compose supports healthchecks. Perhaps we should use that.

https://docs.docker.com/compose/compose-file/compose-file-v3/#healthcheck

Reply:
Hmmm - I do love that we can define these in the compose file, thus outside the existing rigid v2 arch based images, but don't see how we can gate the bash script from continuing.

That is, using this `curl` call we can find out, external to docker, the state of the CHT instance (ie is it still booting/initing) - we don't continue certain bash operations if it's unhealthy.  If we define a health check (thanks for the docs link for easy reference!), it will cause the container(s) to reboot when they fail the check.  This is not the desired outcome as some times boots take a loooooooong time (hence the [ever increasing](https://github.com/medic/cht-core/blob/d6e615fe52f9cd0dd5a6dca54bb835fb07680d1b/scripts/docker-helper/cht-docker-compose.sh#L272) `DEFAULT_SLEEP` duration.

Is there a way we can leverage the health check to not reboot, help bash find out state and thus reduce code in the helper script?